WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift

```diff
@@ -258,6 +265,20 @@ private extension PaymentCaptureOrchestrator {
     }
 }
 
+private extension PaymentCaptureOrchestrator {
+    enum Constants {
+        static let minimumAmount = NSDecimalNumber(string: "0.5")
+    }
+
+    func isTotalAmountValid(order: Order) -> Bool {
+        guard let orderTotal = currencyFormatter.convertToDecimal(from: order.total) else {
+            return false
+        }
+
+        return orderTotal.compare(Constants.minimumAmount) == .orderedDescending
```

Comment:
```suggestion
        return orderTotal as Decimal >= Constants.minimumAmount as Decimal
```

🔧 I think this improves readability (I've never been able to quickly intuit the `orderedDescending` notation) and the Obj-C -> Swift bridging from `NSDecimalNumber` to `Decimal` takes care of everything. Using the [`decimalValue` property](https://developer.apple.com/documentation/foundation/nsnumber/1407409-decimalvalue) also works, so pick your favourite, I guess. I refreshed my memory with some research and couldn't find any difference between them.

I would also consider making the constant a `Decimal` and doing the cast for `orderTotal` in the `guard` here, to neaten the comparison further.

Reply:
In this case, the APIs of currencyConverter expect NSDecimalNumber, so I'd go with the downcast here, even though, as you mentioned, it is not the neatest way to go.